doc/glossary.rst

```diff
@@ -96,4 +96,35 @@ Glossary of Terms
       term storage rather than copying it into RAM, saving writable memory for
       dynamic data and not the static program code.
 
+   Zephyr downstream module
+      A Zephyr module that the :file:`zephyr` repository default configuration
+      and default manifest file are **not** aware of.
+
+   Zephyr manifest module
+      A Zephyr module which also contains one or more west manifest files.
+
+   Zephyr module
+      A YAML file named :file:`zephyr/module.yml` and all the content it
+      references, which define the extensions to be hooked into the Zephyr
+      infrastructure.
```

Comment:
imo we consider the whole project a Zephyr module when it contains a `zephyr/module.yml` file, and not just the file itself.

The file is just used to identify that this project is a Zephyr module, and the content of the file defines how it should be integrated, as example, is it a DTS, board, soc root ? should it be included in the build flow (CMakeLists/Kconfig) ? should it integrate into twister.

Reply:
@tejlmand I started with your impression, then asked if it still made sense for repositories like project_chip and bacnet-stack, which have substantial content for non-Zephyr platforms and environments. It seems the content not reachable by the Zephyr glue code is beyond any Zephyr-module policies. I just didn't want developers in other Open Source projects to think all of their repository needed to comply with Zephyr policies.

Would you still like this changed?

Reply:
I think we need something in between. Cause I find the current description too limiting.

Personally, I would still refer to the project as a Zephyr module, although not all of the code in there may be used by Zephyr.
Just like https://github.com/zephyrproject-rtos/mbedtls is a Zephyr module, but as the code is developed by ARM, then the code itself will not comply with Zephyr policies.

Reply:
@tejlmand Suggestions welcome, and I can align with a more generic definition. Since we can imply a valid `zephyr/module.yml` from repos with both `zephyr/CMakeLists.txt` and `zephyr/Kconfig` (and/or deprecate it), how about:

"A repository with a valid `zephyr/module.yml`" ?